src/App.tsx

```diff
@@ -15,47 +15,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type SortType = 'NONE' | 'ALPHABET' | 'LENGTH';
+type IsReversed = boolean;
+type IsReset = boolean;
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [sortType, setSortType] = useState<SortType>('NONE');
+  const [isReversed, setReverse] = useState<IsReversed>(false);
+  const [isReset, setReset] = useState<IsReset>(false);
+
+  const isOriginalOrder = !isReversed && sortType === 'NONE';
```

Comment:
Here is the reason for the issue. Try this: 
```suggestion
  const isOriginalOrder = !isReversed || sortType === 'NONE';
```